packages/plugin-npm/sources/NpmSemverResolver.ts

```diff
@@ -70,6 +70,17 @@ export class NpmSemverResolver implements Resolver {
     const manifest = new Manifest();
     manifest.load(registryData.versions[version]);
 
+    // Manually add node-gyp dependency if there is a script using it and not already set
+    const ident = structUtils.parseIdent(`node-gyp`);
+    if (!manifest.dependencies.has(ident.identHash) || !manifest.peerDependencies.has(ident.identHash)) {
+      for (const value of manifest.scripts.values()) {
+        if (value.includes(`node-gyp`)) {
```

Comment:
Please add a comment mentioning the funny logic we discussed (that the npm registry silently adds a `node-gyp rebuid` when it detects a `bindings.gyp`, but only does so in the metadata), since it might be harder to uncover for people unfamiliar with the matter.

Reply:
> [...] and, if it does, add an implicit dependency to node-gyp@* (and print a warning through report.reportWarning). Does that make sense? And we can remove the plugin-node-gyp from the bundle as it won't make sense anymore.

I added the comment, but remembered this. I can add a warning like `The dependency <package-name> needs node-gyp but didn't declare it.`

Is it okay?

Reply:
Yup 👍